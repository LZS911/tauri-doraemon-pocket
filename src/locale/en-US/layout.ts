export default {
  header: {
    customization: {
      title: 'Theme Customization',
      themeLayoutPanel: {
        defaultType: 'Default',
        miniType: 'Mini',
        title: 'Theme Layout',
        subTitle: 'Choose a layout type',
      },
      themeModePanel: {
        title: 'Theme Mode',
        subTitle: 'Choose a theme mode',
        light: 'Light',
        dark: 'Dark',
      },
      colorSchemePanel: {
        title: 'Color Scheme',
        subTitle: 'Choose a color scheme',
        default: 'Default',
        theme1: 'Theme1',
        theme2: 'Theme2',
      },
      fontFamilyPanel: {
        title: 'Font Family',
        subTitle: 'Choose a font family',
        default: 'default',
        inter: 'inter',
        poppins: 'poppins',
        roboto: 'roboto',
        serif: 'serif',
        mono: 'mono',
        publicSans: 'public sans',
      },
    },
    notification: {
      title: 'Notification',
      titleTips: 'Mark as all read',
      viewAll: 'View All',
    },
    userNavigation: {
      profile: {
        title: 'profile',
        editProfile: 'Edit Profile',
        viewProfile: 'View Profile',
        bindEmail: 'Bind Email',
        modifyPassword: 'Modify Password',
        singOut: 'SignOut',
      },
      settings: {
        title: 'settings',
        accountSetting: 'Account Setting',
        permissionSetting: 'Permission Setting',
      },
    },
  },
};
