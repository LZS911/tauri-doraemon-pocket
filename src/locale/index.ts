import zhCN from './zh-CN';
import enUS from './en-US';
import * as i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import LocalStorageWrapper from '../utils/LocalStorageWrapper';
import { Dictionary, TemplateKeyPath } from '../typing/common.type';
import CONSTANT from '../common/constant';

export type I18nKey = TemplateKeyPath<typeof zhCN.translation>;

enum SupportLanguage {
  zhCN = 'zh-CN',
  enUS = 'en-US',
}

i18n.use(initReactI18next).init({
  resources: {
    [SupportLanguage.zhCN]: zhCN,
    [SupportLanguage.enUS]: enUS,
  },
  lng: LocalStorageWrapper.getOrDefault(
    CONSTANT.LANGUAGE_STORAGE_KEY,
    SupportLanguage.zhCN
  ),
  fallbackLng: SupportLanguage.zhCN,
  interpolation: {
    escapeValue: false,
  },
});

const t = (key: I18nKey, dic?: Dictionary) => {
  return i18n.t(key, dic);
};

export { SupportLanguage, t };
export default i18n;
