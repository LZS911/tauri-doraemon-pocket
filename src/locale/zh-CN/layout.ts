export default {
  header: {
    customization: {
      title: '主题定制',
      themeLayoutPanel: {
        defaultType: '默认',
        miniType: '简略',
        title: '布局主题',
        subTitle: '选择布局类型',
      },
      themeModePanel: {
        title: '主题模式',
        subTitle: '选择主题类型',
        light: '明亮',
        dark: '暗色',
      },
      colorSchemePanel: {
        title: '配色方案',
        subTitle: '选择配色方案',
        default: '默认',
        theme1: '主题1',
        theme2: '主题2',
      },
      fontFamilyPanel: {
        title: '字体系列',
        subTitle: '选择字体',
        default: 'default',
        inter: 'inter',
        poppins: 'poppins',
        roboto: 'roboto',
        serif: 'serif',
        mono: 'mono',
        publicSans: 'public sans',
      },
    },
    notification: {
      title: '通知',
      titleTips: '标记为全部已读',
      viewAll: '查看全部',
    },
    userNavigation: {
      profile: {
        title: '简介',
        editProfile: '编辑简介',
        viewProfile: '查看简介',
        bindEmail: '邮箱绑定',
        modifyPassword: '修改密码',
        singOut: '注销',
      },
      settings: {
        title: '设置',
        accountSetting: '账号设置',
        permissionSetting: '权限设置',
      },
    },
  },
};
