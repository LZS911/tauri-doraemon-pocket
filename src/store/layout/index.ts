import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import CONSTANT from '../../common/constant';
export interface ILayoutState {
  isExpandSider: boolean;
  siderWidth: number;
  isDrawerSider: boolean;
}
export const layoutInitialState: ILayoutState = {
  isExpandSider: false,
  siderWidth: CONSTANT.SIDER_WIDTH,
  isDrawerSider: false,
};
const layout = createSlice({
  name: 'layout',
  initialState: layoutInitialState,
  reducers: {
    setIsExpandSider: (state, action: PayloadAction<boolean>) => {
      state.isExpandSider = action.payload;
    },
    changeSiderWidth: (state, action: PayloadAction<number>) => {
      state.siderWidth = action.payload;
    },
    setIsDrawerSider: (state, action: PayloadAction<boolean>) => {
      state.isDrawerSider = action.payload;
    },
    resetLayoutState: () => layoutInitialState,
  },
});
export const {
  setIsExpandSider,
  changeSiderWidth,
  resetLayoutState,
  setIsDrawerSider,
} = layout.actions;
export default layout.reducer;
