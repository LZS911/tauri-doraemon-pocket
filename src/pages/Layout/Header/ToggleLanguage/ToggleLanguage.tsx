import { TranslationOutlined } from '@ant-design/icons';
import { genLanguageOptions } from '.';
import ThemeBase, { IOptionsProps } from '../../../../components/ThemeBase';
import useLanguage from '../../../../customHooks/useLanguage';
import { SupportLanguage } from '../../../../locale';

const ToggleLanguage: React.FC = () => {
  const { changeLanguage, currentLanguage } = useLanguage();

  const handleClickLanguageItem: IOptionsProps<SupportLanguage>['onChange'] = (
    lang
  ) => {
    lang && changeLanguage(lang);
  };

  return (
    <ThemeBase.Options
      value={currentLanguage}
      onChange={handleClickLanguageItem}
      list={genLanguageOptions()}
    >
      <ThemeBase.Icon
        data-testid="open-language-options"
        icon={<TranslationOutlined className="text-[1rem]" />}
        className="mr-2 bg-slate-100 dark:bg-black"
      />
    </ThemeBase.Options>
  );
};

export default ToggleLanguage;
