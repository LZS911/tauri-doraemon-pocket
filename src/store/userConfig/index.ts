import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import CONSTANT from '../../common/constant';
import {
  ThemeModeEnum,
  ColorSchemeEnum,
  FontFamilyEnum,
} from '../../common/enum';
import { SupportLanguage } from '../../locale';
import LocalStorageWrapper from '../../utils/LocalStorageWrapper';
export interface IUserConfigState {
  username: string;
  emailAddress: string;
  userProfile: string;
  isLogin: boolean;
  token: string;
  currentThemeMode: ThemeModeEnum;
  currentColorScheme: ColorSchemeEnum;
  lang: SupportLanguage;
  fontFamily: FontFamilyEnum;
}
export type IUserLoginInfo = Pick<
  IUserConfigState,
  'username' | 'emailAddress' | 'userProfile'
>;
export type IUserLoginState = Pick<IUserConfigState, 'token' | 'isLogin'>;
const localToken = LocalStorageWrapper.get(CONSTANT.TOKEN);
export const userConfigInitialState: IUserConfigState = {
  username: '',
  emailAddress: '',
  userProfile: '',
  isLogin: !!localToken,
  token: localToken ?? '',
  currentThemeMode: LocalStorageWrapper.getOrDefault(
    CONSTANT.THEME_MODE,
    ThemeModeEnum.Light
  ),
  currentColorScheme: LocalStorageWrapper.getOrDefault(
    CONSTANT.COLOR_SCHEME,
    ColorSchemeEnum.Blue
  ),
  lang: LocalStorageWrapper.getOrDefault(
    CONSTANT.LANGUAGE_STORAGE_KEY,
    SupportLanguage.zhCN
  ),
  fontFamily: LocalStorageWrapper.getOrDefault(
    CONSTANT.FONT_FAMILY,
    FontFamilyEnum.Inter
  ),
};
const userConfig = createSlice({
  name: 'userConfig',
  initialState: userConfigInitialState,
  reducers: {
    setLoginUserInfo: (state, action: PayloadAction<IUserLoginInfo>) => {
      state.username = action.payload.username;
      state.emailAddress = action.payload.emailAddress;
      state.userProfile = action.payload.userProfile;
    },
    setLoginState: (state, action: PayloadAction<IUserLoginState>) => {
      state.isLogin = action.payload.isLogin;
      state.token = action.payload.token;
      localStorage.setItem(CONSTANT.TOKEN, action.payload.token);
    },
    setCurrentThemeMode: (state, action: PayloadAction<ThemeModeEnum>) => {
      state.currentThemeMode = action.payload;
      LocalStorageWrapper.set(CONSTANT.THEME_MODE, action.payload);
    },
    setCurrentColorScheme: (state, action: PayloadAction<ColorSchemeEnum>) => {
      state.currentColorScheme = action.payload;
      LocalStorageWrapper.set(CONSTANT.COLOR_SCHEME, action.payload);
    },
    setCurrentLang: (state, action: PayloadAction<SupportLanguage>) => {
      state.lang = action.payload;
      LocalStorageWrapper.set(CONSTANT.LANGUAGE_STORAGE_KEY, action.payload);
    },
    setFontFamily: (state, action: PayloadAction<FontFamilyEnum>) => {
      state.fontFamily = action.payload;
      LocalStorageWrapper.set(CONSTANT.FONT_FAMILY, action.payload);
    },
  },
});
export const {
  setLoginUserInfo,
  setLoginState,
  setCurrentThemeMode,
  setCurrentColorScheme,
  setCurrentLang,
  setFontFamily,
} = userConfig.actions;
export default userConfig.reducer;
