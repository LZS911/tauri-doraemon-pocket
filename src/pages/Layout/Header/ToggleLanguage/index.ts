import i18n, { SupportLanguage } from '../../../../locale';
import ToggleLanguage from './ToggleLanguage';

export default ToggleLanguage;

export const genLanguageOptions = () => {
  return [
    {
      key: 'zhCN',
      value: SupportLanguage.zhCN,
      text: i18n.t('common.chinese'),
    },
    {
      key: 'enUS',
      value: SupportLanguage.enUS,
      text: i18n.t('common.english'),
    },
  ];
};
