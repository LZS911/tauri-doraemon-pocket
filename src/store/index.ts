import { configureStore, Store } from '@reduxjs/toolkit';
import userConfig from './userConfig';
import layout from './layout';
const store: Store = configureStore({
  reducer: {
    userConfig,
    layout,
  },
});
export type IReduxState = ReturnType<typeof store.getState>;
export default store;
