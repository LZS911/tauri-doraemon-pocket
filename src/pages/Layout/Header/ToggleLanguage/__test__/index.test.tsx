import { fireEvent, render, screen } from '@testing-library/react';
import { useDispatch, useSelector } from 'react-redux';
import { SupportLanguage } from '../../../../../locale';
import { layoutInitialState } from '../../../../../store/layout';
import { userConfigInitialState } from '../../../../../store/userConfig';
import ToggleLanguage from '../index';

jest.mock('react-redux', () => {
  return {
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn(),
    useDispatch: jest.fn(),
  };
});

describe('test Customization', () => {
  const useSelectorMock: jest.Mock = useSelector as jest.Mock;
  const useDispatchMock: jest.Mock = useDispatch as jest.Mock;
  const dispatchMock = jest.fn();

  beforeEach(() => {
    useDispatchMock.mockImplementation(() => dispatchMock);
    useSelectorMock.mockImplementation((selector) => {
      return selector({
        userConfig: userConfigInitialState,
        layout: layoutInitialState,
      });
    });
  });
  afterEach(() => {
    jest.clearAllMocks();
    jest.clearAllTimers();
  });

  test('should match snapshot', () => {
    const { container } = render(<ToggleLanguage />);
    fireEvent.click(screen.getByTestId('open-language-options'));
    expect(container).toMatchSnapshot();
  });

  test('should modify language when click option', () => {
    render(<ToggleLanguage />);
    fireEvent.click(screen.getByTestId('open-language-options'));
    expect(dispatchMock).toBeCalledTimes(0);
    fireEvent.click(screen.getByText('common.english'));
    expect(dispatchMock).toBeCalledTimes(1);
    expect(dispatchMock).toBeCalledWith({
      payload: SupportLanguage.enUS,
      type: 'userConfig/setCurrentLang',
    });
  });
});
