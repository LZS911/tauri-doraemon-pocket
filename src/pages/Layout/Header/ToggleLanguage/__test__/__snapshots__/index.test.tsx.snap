// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Customization should match snapshot 1`] = `
<div>
  <div
    class="ant-popover-open"
  >
    <div
      class="hover:bg-slate-200 dark:hover:bg-slate-400 after:water-wave-hide active:after:water-wave-show mr-2 bg-slate-100 dark:bg-black transparent relative flex cursor-pointer items-center justify-center rounded-md bg-transparent p-2 transition-colors dark:text-white"
      data-testid="open-language-options"
    >
      <span
        aria-label="translation"
        class="anticon anticon-translation text-[1rem]"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="translation"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <defs>
            <style />
          </defs>
          <path
            d="M140 188h584v164h76V144c0-17.7-14.3-32-32-32H96c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h544v-76H140V188z"
          />
          <path
            d="M414.3 256h-60.6c-3.4 0-6.4 2.2-7.6 5.4L219 629.4c-.3.8-.4 1.7-.4 2.6 0 4.4 3.6 8 8 8h55.1c3.4 0 6.4-2.2 7.6-5.4L322 540h196.2L422 261.4a8.42 8.42 0 00-7.7-5.4zm12.4 228h-85.5L384 360.2 426.7 484zM936 528H800v-93c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v93H592c-13.3 0-24 10.7-24 24v176c0 13.3 10.7 24 24 24h136v152c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V752h136c13.3 0 24-10.7 24-24V552c0-13.3-10.7-24-24-24zM728 680h-88v-80h88v80zm160 0h-88v-80h88v80z"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
